src/ts.c

```diff
@@ -293,6 +314,15 @@ void add_ts_packet_data(unsigned char *buf, mumudvb_ts_packet_t *pkt, int data_l
 		}
 		else
 		{
+			if(pkt->status_partial==PARTIAL_HEADER)
+			{
+				//read up to the first 3 bytes, to get the packet length
+				memcpy(pkt->data_partial+pkt->len_partial,buf,3-pkt->len_partial);
```

Comment:
We need to check than (3-len_partial) > 0
And that we have enough data to read the length
Because of transmission errors, the TS data can be quite garbage sometimes